packages/devtools_app/lib/src/service/vm_service_wrapper.dart

```diff
@@ -284,6 +289,14 @@ class VmServiceWrapper implements VmService {
     int? offset,
     int? count,
   }) {
+    final cachedObj = fakeServiceCache.getObject(
```

Comment:
should we really be adding this logic in vm_service_wrapper? this seems like it should belong in `JsonToServiceCache` or some other helper / manager.

Reply:
From what I understand, this is complicated by the fact that the widgets are hardcoded to depend on this global. So changing that dependency, (e.g. passing in the service manager instead of accessing it through the global, so that we could pass a fake one when we are using it for different purposes) would take a lot of work in refactoring.

Reply:
I figured this would also give us support for adding data to the cache for use by other widgets with minimal overhead. `Map.contains` should be orders of magnitude less expensive than an actual service request, so there shouldn't be any noticeable performance difference.